a3p-integration/proposals/z:acceptance/test-lib/sync-tools.test.js

```diff
@@ -411,3 +444,118 @@ test.serial('wait until invitation recevied', async t => {
 
   await t.notThrowsAsync(waitP);
 });
+
+test.serial('wait election result: question handle is adhered', async t => {
```

Comment:
These tests all verify the happy path. Please add a test for no resolution after `maxRetries`.

Reply:
The reason I haven’t added any tests for `waitP` rejecting is that in order for the test to pass, the condition passed to `retryUntilCondition` must return false until `setTimeout` updates the state returned from `follow`. 

On a second thought, there’s no way to make sure enough retries happen until state is updated to expected one since the retry information is only logged. So it’s probably a good practice to add a test for no resolution case. Thanks.

Reply:
You can run a test with small maxRetries and small retryIntervalMs, right?

Reply:
Yep, it's resolved here https://github.com/Agoric/agoric-sdk/pull/10385/commits/7b207b470d3995d54cfd2f9928837dee50257730